examples/routing_service/routing_service_file_adapter/cpp/FileAdapter.hpp

```diff
@@ -0,0 +1,49 @@
+/*****************************************************************************/
+/*         (c) Copyright, Real-Time Innovations, All rights reserved.        */
+/*                                                                           */
+/*         Permission to modify and use for internal purposes granted.       */
+/* This software is provided "as is", without warranty, express or implied.  */
+/*                                                                           */
+/*****************************************************************************/
+#ifndef FILEADAPTER_HPP
+#define FILEADAPTER_HPP
+
+#include <dds/core/corefwd.hpp>
+#include <dds/core/macros.hpp>
+#include <rti/routing/PropertySet.hpp>
+#include <rti/routing/RoutingService.hpp>
+#include <rti/routing/adapter/AdapterPlugin.hpp>
+
+namespace rti { namespace community { namespace examples {
+class FileAdapter : public rti::routing::adapter::AdapterPlugin {
+public:
+    FileAdapter(rti::routing::PropertySet &){};
```

Comment:
Make it `explicit`.

[C.46: By default, declare single-argument constructors explicit](http://intranet:8000//local/preship/cpp_core_guidelines/sphinx/_build/html/CppCoreGuidelines.html#Rc-explicit)

Remove the trailing `;`

Reply:
Done. I removed the {} instead, since this is just a declaration.

Reply:
You can't remove the `{}` can you? Won't RS call that constructor? If you don't need it at all, delete the whole constructor, don't declare it and not define it.

Reply:
You are right. I ended up defining and empty constructor body. We can remove that if needed.